### ISSUE TITLE
Making World Generation look more like vanilla

### ISSUE THREAD

**Issue by Titaniumtown** [cuberite/cuberite#4759]
So, I was looking into how one could make cuberite world gen more like vanilla. I found a project (https://github.com/Cubitect/cubiomes) that aims to port vanilla like world gen to C. It has various options, including toggling between different versions of minecraft's generation. It even seems to have 1.16 generation figured out. 

Thoughts?

Edit: it seems after further reading, it doesn't generate a world file. But cords where biomes and structures are.

**Comment by Titaniumtown** [comment]
bump, is this an avenue that would be worth looking into?

**Comment by bearbin** [comment]
Please don't 'bump' issues. That looks like a relatively interesting project, but it's only going to get integrated if somebody turns up and actually wants to do it.

**Comment by Titaniumtown** [comment]
ok sorry. I'm looking into it now.

**Comment by Titaniumtown** [comment]
I don't think I have the know how to do this. lol

**Comment by bearbin** [comment]
From my cursory look at it, it seems to only do the biome generation,
and structure placement. The actual 'blocks' don't seem to be something
it's capable of doing. Luckily the cuberite generator is similarly
modular, so to integrate it all that would be necessary would be to
write a biome generator/finisher including it as a library and calling
into it at the relevant places.

I don't have any particular desire to do it myself, but it seems like it
shouldn't be too difficult (although problems will likely pop up...)

A bigger issue would be licensing. cubiomes is GPL licensed which
precludes its inclusion in cuberite with the Apache license. Any
combination of cuberite with cubiomes would end up as GPL as the GPL is
viral. I have no desire to make Cuberite GPL and resultantly impose
additional restrictions on users; neither would there be a desire to
implement some sort of complicated license-avoiding system like whatever
the current successor to CraftBukkit has.

Unfortunately the poor choice of license on their part puts a stop to
things.

**Comment by Titaniumtown** [comment]
Aww that sucks. I'll close the issue then :( It sucks that licensing gets in the way of things.

**Comment by bearbin** [comment]
I have submitted an issue on cubiomes about the license, let's see what
they say: 

https://github.com/Cubitect/cubiomes/issues/34

**Comment by Titaniumtown** [comment]
oooh that would be neat

**Comment by Titaniumtown** [comment]
keep me posted!

**Comment by Titaniumtown** [comment]
Now that the License has been changed. Work can be started on implemented cubiomes in cuberite, right?

**Comment by madmaxoft** [comment]
Yeah, but I guess you'll soon find out that Cuberite already uses a biome generator that is pretty much identical to vanilla, except for perhaps some constants.

I see more usefulness in their structure positioning code, that could be of interest.

**Comment by Titaniumtown** [comment]
oh ok. got it. I'll look into this when I have some time! :)

**Comment by PureTryOut** [comment]
Yeah the current world generation makes for some... interesting... worlds. For example, check seed 290400380 at X -44 and Z 36. There is a 214 block high mountain in a Savanna Plateau biome with a maximum surface area of about 20 by 20. I mean, yes this biome should have some mountains but this is ridiculous lol.
